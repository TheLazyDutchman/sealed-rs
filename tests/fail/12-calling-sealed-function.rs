mod inner {
    use sealed::sealed;

    #[sealed(partial)]
    pub trait A {
        #[seal(uncallable)]
        fn sealed() {}

        fn unsealed();
    }
}

use inner::A;

struct B;

impl A for B {
    fn unsealed() {}
}

fn main() {
    B::sealed();
    B::sealed(Token);
    B::unsealed();
}
