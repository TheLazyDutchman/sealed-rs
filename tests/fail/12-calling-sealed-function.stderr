error[E0425]: cannot find value `Token` in this scope
  --> tests/fail/12-calling-sealed-function.rs:23:15
   |
23 |     B::sealed(Token);
   |               ^^^^^ not found in this scope
   |
help: consider importing this unit struct
   |
13 | use crate::inner::__seal_a::Token;
   |

error[E0061]: this function takes 1 argument but 0 arguments were supplied
  --> tests/fail/12-calling-sealed-function.rs:22:5
   |
22 |     B::sealed();
   |     ^^^^^^^^^-- supplied 0 arguments
   |     |
   |     expected 1 argument
   |
note: associated function defined here
  --> tests/fail/12-calling-sealed-function.rs:7:12
   |
4  | /     #[sealed(partial)]
5  | |     pub trait A {
6  | |         #[seal(uncallable)]
7  | |         fn sealed() {}
   | |____________^^^^^-
